evm/src/coverage/visitor.rs

```diff
@@ -1,24 +1,27 @@
-use super::{BranchKind, CoverageItem, ItemAnchor, SourceLocation};
+use super::{CoverageItem, ItemAnchor, SourceLocation};
 use ethers::{
-    prelude::sourcemap::SourceMap,
+    prelude::{sourcemap::SourceMap, Bytes},
     solc::artifacts::ast::{self, Ast, Node, NodeType},
 };
-use std::collections::HashMap;
+use revm::{opcode, spec_opcode_gas, SpecId};
+use std::{
+    borrow::Cow,
+    collections::{BTreeMap, HashMap},
+};
 use tracing::warn;
 
 #[derive(Debug, Default, Clone)]
-pub struct Visitor {
-    /// The source ID containing the AST being walked.
-    source_id: u32,
+pub struct Visitor<'a> {
     /// The source code that contains the AST being walked.
     source: String,
     /// Source maps for this specific source file, keyed by the contract name.
     source_maps: HashMap<String, SourceMap>,
+    /// Bytecodes for this specific source file, keyed by the contract name.
+    bytecodes: HashMap<String, Cow<'a, Bytes>>,
```

Comment:
Bytes is cheaply cloneable because it's basically just an AtomicPointer and a `* u8` pointer to the data, so no need to use cow, I think

Reply:
the cow is from ethers-rs :smile:

Reply:
oh, maybe I should change that, I probably introduced this by copy-pasting the other functions...